test/Altinn.Platform.Events.Functions.Tests/TestingServices/CertificateResolverServiceTests.cs

```diff
@@ -0,0 +1,78 @@
+using Altinn.Platform.Events.Functions.Configuration;
+using Altinn.Platform.Events.Functions.Services;
+using Altinn.Platform.Events.Functions.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using Xunit;
+
+namespace Altinn.Platform.Events.Functions.Tests.Services
+{
+    public class CertificateResolverServiceTests
+    {
+        [Fact]
+        public async Task GetCertificateAsync_ShouldRetrieveCertificateFromKeyVault()
+        {
+            // Arrange
+            var loggerMock = new Mock<ILogger<ICertificateResolverService>>();
+            var certificateResolverSettings = Options.Create(new CertificateResolverSettings { CacheCertLifetimeInSeconds = 3600 }); // 1 hour cache lifetime
+            var keyVaultServiceMock = new Mock<IKeyVaultService>();
+            var keyVaultSettings = Options.Create(new KeyVaultSettings { KeyVaultURI = "https://altinn-at21-kv.vault.azure.net", PlatformCertSecretId = "platform-cert" });
```

Comment:
I would invent a URL for the test to make it more obvious that it's not an integration test.